core/src/main/java/com/sequenceiq/cloudbreak/service/stackpatch/DisableRegionForFluentdPatchService.java

```diff
@@ -0,0 +1,138 @@
+import com.sequenceiq.cloudbreak.util.CompressUtil;
+
+@Service
+public class DisableRegionForFluentdPatchService extends AbstractTelemetryPatchService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DisableRegionForFluentdPatchService.class);
+
+    @Inject
+    private CompressUtil compressUtil;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private ClusterBootstrapper clusterBootstrapper;
+
+    @Inject
+    private StackImageService stackImageService;
+
+    @Inject
+    private DisableRegionForFluentdPatchConfig disableRegionForFluentdPatchConfig;
+
+    private ModuleDescriptor.Version affectedVersion;
+
+    @PostConstruct
+    public void init() {
+        affectedVersion = ModuleDescriptor.Version.parse(disableRegionForFluentdPatchConfig.getAffectedVersionFrom());
+    }
+
+    @Override
+    boolean doApply(Stack stack) throws ExistingStackPatchApplyException {
+        try {
+            byte[] currentSaltState = getCurrentSaltStateStack(stack);
+            List<String> saltStateDefinitions = Arrays.asList("salt-common", "salt");
+            List<String> loggingAgentSaltStateDef = List.of("/salt/fluent");
+            byte[] fluentSaltStateConfig = compressUtil.generateCompressedOutputFromFolders(saltStateDefinitions, loggingAgentSaltStateDef);
+            boolean loggingAgentContentMatches = compressUtil.compareCompressedContent(currentSaltState, fluentSaltStateConfig, loggingAgentSaltStateDef);
+            if (!loggingAgentContentMatches) {
+                Set<InstanceMetaData> instanceMetaDataSet = instanceMetaDataService.findNotTerminatedAndNotZombieForStack(stack.getId());
+                List<GatewayConfig> gatewayConfigs = gatewayConfigService.getAllGatewayConfigs(stack);
+                ClusterDeletionBasedExitCriteriaModel exitModel = ClusterDeletionBasedExitCriteriaModel.nonCancellableModel();
+                Set<Node> availableNodes = getAvailableNodes(instanceMetaDataSet, gatewayConfigs, exitModel);
+                if (CollectionUtils.isEmpty(availableNodes)) {
+                    LOGGER.info("Not found any available nodes for patch, stack: " + stack.getName());
+                    return false;
+                } else {
+                    getTelemetryOrchestrator().updateAndRestartTelemetryService(fluentSaltStateConfig, "fluent", "fluent.init",
+                            gatewayConfigs, availableNodes, exitModel);
+                    byte[] newFullSaltState = compressUtil.updateCompressedOutputFolders(saltStateDefinitions, loggingAgentSaltStateDef, currentSaltState);
+                    clusterBootstrapper.updateSaltComponent(stack, newFullSaltState);
+                    LOGGER.debug("Disable region for fluentd has been successfully finished for stack {}", stack.getName());
```

Comment:
What about the unresponsive nodes? Is a partial success enough?

Reply:
state is stored, so on next highstate it will be fine, so that's good enough